src/lib.rs

```diff
@@ -20,6 +20,34 @@
 //! upgrade policy, where in a later 1.x version, we will raise the minimum
 //! required Rust version.
 
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(not(feature = "std"))]
+extern crate hashmap_core;
+
+#[cfg(not(feature = "std"))]
+mod std {
+    pub use core::*;
+
+    pub mod vec {
+        pub use alloc::vec::*;
+    }
+
+    pub mod boxed {
+        pub use alloc::boxed::*;
+    }
+
+    pub mod collections {
+        pub mod hash_map {
+            pub use hashmap_core::map::*;
```

Comment:
I just realized this changes our default `RandomState` type, normally defined only in `std`, which is also a breaking change.

Reply:
Yes, I think hashmap_core should be dropped from this PR, because it only adds this, which we don't want.

We'd like to not have a default hasher, that should work, and not have IndexMap::new() at all when we are no-std.